Document the Weekly Release Line (#3512)
jenkins-infra/jenkins.io: content/download/weekly/index.adoc

---
layout: simplepage
title: Weekly Release Line
---
:toc:

The Jenkins _Weekly Release Line_ delivers bug fixes and new features rapidly to users and plugin developers who need them.
These releases are generally delivered on a weekly cadence, hence the name.
For more conservative users, there is also a 
link:/download/lts/[Long-Term Support (LTS)] line.

## Model

The "weekly" release cadence is **aspirational**.
Generally we target releasing Jenkins every week, but sometimes we do out-of-order releases.

> To get notifications about new releases, it is possible to subscribe to link:https://www.jenkins.io/changelog/rss.xml[releases RSS feed].

Reviewer: or github releases?

Author: Right now we do not have automation for them, so I would not document it until the process is fully automated and GitHub Releases are included into our release Pipeline

Reviewer: it's better than RSS imo even if not fully automated but yeah would be great to finish that off

Conditions for new releases of the _Weekly Release Line_:

> * **Common releases** with all features, improvements and bug fixes integrated since the last release.

Reviewer 2: ```suggestion
* **Regular releases** with all features, improvements and bug fixes integrated since the last release.
```

> * **Common releases** with all features, improvements and, and bugfixes integrated since the last release.

Reviewer 3: Not sure whether latest fits better.

```suggestion
* **Common releases** with all features, improvements and bug fixes integrated since the last release.
```

** Currently these releases happen on a weekly basis on Tuesdays.
* **Security releases** for the Jenkins core (see the link:/security[Jenkins Security] page).
** We coordinate Jenkins Weekly and LTS releases so that the vulnerabilities are fixed at the same time in all release lines.

>    Jenkins security team sends notifications about incoming security releases to link:/security/#advisories[Jenkins Security Advisory Channels].

Reviewer: ```suggestion
   Jenkins security team sends notifications about upcoming security releases to link:/security/#advisories[Jenkins Security Advisory Channels].
```

> ** Usually these releases happen on Wednesdays.

Reviewer 2: ```suggestion
** Usually these releases happen on Wednesdays.
   If we release security updates on Wednesday, there will be no regular release the day before.
```

* **Out-of-order releases** in the case of major regressions and/or other reasons.

Weekly releases are managed by Jenkins core maintainers
who review the suggested pull requests, integrate changes and deliver the releases.
The team also manages link:/changelog[changelogs] and the LTS backporting process.
See link:https://github.com/jenkinsci/jenkins/blob/master/docs/MAINTAINERS.adoc[this page] for more information about Jenkins core maintenance.

## No Backporting

> Unlike for the LTS baseline,

Reviewer 2: ```suggestion
Unlike for the LTS line,
```

there is NO backporting process for the _Weekly Release Line_.

> If a new release is needed to fix a discovered regression,
> we just release a new Jenkins weekly version. 

Reviewer 2: ```suggestion
Bug fixes, including fixes for regressions will be integrated towards the next release.
While that release may be expedited (see out-of-order releases above), there won't be patches for previous releases.
```

## Switching Between Release Lines

> It is possible to switch between Jenkins Weekly and LTS release line when needed.

Reviewer 3: ```suggestion
It is possible to switch between Jenkins Weekly and LTS release lines when needed.
```

---
layout: simplepage
title: Weekly Release Line
---
:toc:

The Jenkins _Weekly Release Line_ delivers bug fixes and new features rapidly to users and plugin developers who need them.
These releases are generally delivered on a weekly cadence, hence the name.
For more conservative users, there is also a 
link:/download/lts/[Long-Term Support (LTS)] line.

## Model

The "weekly" release cadence is **aspirational**.
Generally we target releasing Jenkins every week, but sometimes we do out-of-order releases.
To get notifications about new releases, it is possible to subscribe to link:https://www.jenkins.io/changelog/rss.xml[releases RSS feed].

Conditions for new releases of the _Weekly Release Line_:

> * **Regular releases** with all features, improvements and bug fixes integrated since the last release.

Reviewer 2: ```suggestion
* **Regular releases** with all features, improvements, and bug fixes integrated since the last release.
```

Personal preference. Sadly, the style guide does not mention Oxford comma.

Author: @MarkEWaite was not it you who also put +1 on the suggestion by @MRamonLeon to remove it? I am +1 for documenting the recommendations for that in the style guide

Reviewer 2: The original had a duplicate "and"which was the major problem.

Author: Ah right

> ** Currently these releases happen on a weekly basis on Tuesdays.

Reviewer 2: As a personal preference would make this a dl so we don't need nested list items, that seems excessive.

See e.g. https://www.jenkins.io/security/for-maintainers/#support-for-maintainers or https://www.jenkins.io/security/#learn-more for examples.

Author: I think the list needs a bit more entries to get seriously impacted by look&feel. If we extend the documentation in this section any more, I commit to update the layout

* **Security releases** for the Jenkins core (see the link:/security[Jenkins Security] page).
** We coordinate Jenkins Weekly and LTS releases so that the vulnerabilities are fixed at the same time in all release lines.
   Jenkins security team sends notifications about upcoming security releases to link:/security/#advisories[Jenkins Security Advisory Channels].
** Usually these releases happen on Wednesdays.

>    If we release security updates on Wednesday, there will be no regular release the day before.

Reviewer 2: ```suggestion
   If we release security updates on Wednesday, there will be no regular release the day before.
   link:/security/scheduling/[Learn more about how we schedule security advisories].
```

* **Out-of-order releases** in the case of major regressions and/or other reasons.

Weekly releases are managed by Jenkins core maintainers
who review the suggested pull requests, integrate changes and deliver the releases.
The team also manages link:/changelog[changelogs] and the LTS backporting process.
See link:https://github.com/jenkinsci/jenkins/blob/master/docs/MAINTAINERS.adoc[this page] for more information about Jenkins core maintenance.

## No Backporting

Unlike for the LTS line,

> there is NO backporting process for the _Weekly Release Line_.

Reviewer 2: ```suggestion
there is no backporting process for the _Weekly Release Line_.
```

IMO allcaps is weird on the web when you could use formatting to emphasize. (I also disagree that this needs special emphasize given it has its own section header that says basically the same thing, so I left it out).
